src/hb-subset-input.cc

```diff
@@ -45,15 +45,22 @@ hb_subset_input_create_or_fail (void)
   if (unlikely (!input))
     return nullptr;
 
-  input->unicodes = hb_set_create ();
-  input->glyphs = hb_set_create ();
-  input->name_ids = hb_set_create ();
+  if (unlikely ((input->unicodes = hb_set_create ())->in_error ()))
+    return nullptr;
+  if (unlikely ((input->glyphs = hb_set_create ())->in_error ()))
+    return nullptr;
+  if (unlikely ((input->name_ids = hb_set_create ())->in_error ()))
+    return nullptr;
   hb_set_add_range (input->name_ids, 0, 6);
-  input->name_languages = hb_set_create ();
+  if (unlikely ((input->name_languages = hb_set_create ())->in_error ()))
+    return nullptr;
   hb_set_add (input->name_languages, 0x0409);
-  input->layout_features = hb_set_create ();
-  input->drop_tables = hb_set_create ();
-  input->no_subset_tables = hb_set_create ();
+  if (unlikely ((input->layout_features = hb_set_create ())->in_error ()))
+    return nullptr;
+  if (unlikely ((input->drop_tables = hb_set_create ())->in_error ()))
+    return nullptr;
+  if (unlikely ((input->no_subset_tables = hb_set_create ())->in_error ()))
+    return nullptr;
```

Comment:
Doesn't this block leak earlier objects?

Much easier when we switch to a common API for all sets.

Reply:
Yes, good catch. I'll fix.